### ISSUE TITLE
Way to reconnect to camera after permission grant on Android M.

### ISSUE THREAD

**Issue by localhostEU** [dm77/barcodescanner#137]
Is there any way to do so? startCamera() doesn't work

**Comment by pierrre** [comment]
Currently in `BarcodeScannerView`, the `CameraPreview` and `IViewFinder` are created during the constructor call.
You need to create a new `BarcodeScannerView`/`ZXingScannerView`.
You can:
- create and attach a new view
- replace/refresh the fragment in your activity
- restart the same activity

**Comment by localhostEU** [comment]
Thanks for the reply, solved it but this dirty hack:
```
getFragmentManager().beginTransaction()
                    .detach(this)
                    .attach(this)
                    .commitAllowingStateLoss();
```

**Comment by pierrre** [comment]
Thank you.

**Comment by dm77** [comment]
Is it possible to instantiate your fragment or view after getting the permission?

**Comment by pierrre** [comment]
It seems to be fixed in 1.8.4.
Can you confirm?

**Comment by dm77** [comment]
I have updated the samples to show how to request for permission. The library itself is not responsible for handling camera permissions. But the samples in 1.8.4 should help you :)
